bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTBridgeHandler.java

```diff
@@ -0,0 +1,300 @@
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.bn.coders.IASN1PreparedElement;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.GsonBuilder;
+
+import net.heberling.ismart.asn1.v1_1.Message;
+import net.heberling.ismart.asn1.v1_1.MessageCoder;
+import net.heberling.ismart.asn1.v1_1.entity.AlarmSwitch;
+import net.heberling.ismart.asn1.v1_1.entity.AlarmSwitchReq;
+import net.heberling.ismart.asn1.v1_1.entity.MP_AlarmSettingType;
+import net.heberling.ismart.asn1.v1_1.entity.MP_UserLoggingInReq;
+import net.heberling.ismart.asn1.v1_1.entity.MP_UserLoggingInResp;
+import net.heberling.ismart.asn1.v1_1.entity.MessageListReq;
+import net.heberling.ismart.asn1.v1_1.entity.MessageListResp;
+import net.heberling.ismart.asn1.v1_1.entity.StartEndNumber;
+import net.heberling.ismart.asn1.v1_1.entity.VinInfo;
+
+/**
+ * The {@link SAICiSMARTBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Heberling - Initial contribution
+ */
+@NonNullByDefault
+public class SAICiSMARTBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SAICiSMARTBridgeHandler.class);
+
+    private @Nullable SAICiSMARTBridgeConfiguration config;
+
+    private @Nullable String uid;
+
+    private @Nullable String token;
+
+    private @Nullable Collection<VinInfo> vinList;
+    private HttpClient httpClient;
+    private @Nullable Future<?> pollingJob;
+
+    private SAICTranslationProvider i18nProvider;
+
+    public SAICiSMARTBridgeHandler(SAICTranslationProvider i18nProvider, Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.i18nProvider = i18nProvider;
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands available
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SAICiSMARTBridgeConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Validate configuration
+        if (this.config.username.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, i18nProvider
```

Comment:
Why do you need to create a custom i18n provider? 

https://www.openhab.org/docs/developer/utils/i18n.html#using-custom-keys shows an example using 

`updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "@text/offline.communication-error");' maybe that is a lot simpler?

Reply:
Nice. I missed that trick.  I have removed the custom i18n provider.